### ISSUE TITLE
MAINT, TST: pytest issues?

### ISSUE THREAD

**Issue by tylerjereddy** [darshan-hpc/darshan#723]
The release was quite recent and now shows this in CI per gh-722

```
 Using pytest.skip outside of a test will skip the entire module. If that's your intention, pass allow_module_level=True. If you want to skip a specific test or an entire class, use the @pytest.mark.skip or @pytest.mark.skipif decorators.
```

My current suggestion is just to pin this for a while, since we do some fancy stuff with test skipping in `darshan/log_utils.py` and I wouldn't want to accidentally skip a bunch of tests, etc.

**Comment by tylerjereddy** [comment]
I'm not able to reproduce locally yet though.. I'll try a restart on that job as well.

**Comment by tylerjereddy** [comment]
Actually, I don't think it can be `pytest` `7.1.2` that is to blame--test suite already passing with that version in gh-718.

**Comment by tylerjereddy** [comment]
Reproducible locally with the `act` tool for `platform linux -- Python 3.8.12, pytest-7.0.1, pluggy-1.0.0`.

A bit of a stretch, but I do know that some base docker images changed recently because of the `git` security vulnerability: https://github.blog/2022-04-12-git-security-vulnerability-announced/

**Comment by tylerjereddy** [comment]
The problem is the usage of `get_log_path()` in a `pytest` parametrization decorator.

From the `get_log_path()` docs we wrote:

> This function should never be used in a pytest decorator/parametrization mark. While it is possible for the function to retrieve log file paths in such scenarios, the imperative skip is not tolerated at collection time when the logs repo is absent.

This was a case where the logs in question were in the main repo so we didn't observe the issue, but when I expanded the test to include logs repo data, the fundamental reason we shouldn't do this was exposed. So, I just need to shift the `get_log_path()` calls to within the body of the test.
